src/wallet/signer.ts

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable import/max-dependencies -- These imports are required for Signer */
+import { BigNumber } from "bignumber.js";
+import { flatMap } from "lodash";
+import { decodeAccountID } from "ripple-address-codec";
+import {
+  encode,
+  decode,
+  encodeForSigning,
+  encodeForSigningClaim,
+} from "ripple-binary-codec";
+import {
+  sign as signWithKeypair,
+  verify as verifySignature,
+} from "ripple-keypairs";
+
+import { ValidationError } from "../common/errors";
+import { SignedTransaction } from "../common/types/objects";
+import { Signer } from "../models/common";
+import { Transaction } from "../models/transactions";
+import { verifyBaseTransaction } from "../models/transactions/common";
+import { computeBinaryTransactionHash } from "../utils";
+import Wallet from "../Wallet";
+
+/**
+ * Sign uses your wallet to cryptographically sign your transaction which proves
+ * that you are the one issuing this transaction.
+ *
+ * @param wallet - A Wallet that holds your cryptographic keys.
+ * @param tx - The Transaction that is being signed.
+ * @returns A signed Transaction and a Transaction id that corresponds to the pre-signed Transaction.
+ */
+function sign(wallet: Wallet, tx: Transaction): SignedTransaction {
```

Comment:
Rearrange the methods so that all the private helpers are at the bottom and the exported methods are at the top (so it's easier to follow what's going on).